module/data/activity/save-data.mjs

```diff
@@ -82,4 +98,34 @@ export default class SaveActivityData extends BaseActivityData {
     if ( !this.damage.onSave ) this.damage.onSave = this.isSpell && (this.item.system.level === 0) ? "none" : "half";
     if ( !this.save.dc.calculation ) this.save.dc.calculation = this.isSpell ? "spellcasting" : "custom";
   }
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  prepareFinalData(rollData) {
+    rollData ??= this.getRollData({ deterministic: true });
+    super.prepareFinalData(rollData);
+    this.prepareDamageLabel(this.damage.parts, rollData);
+
+    let ability;
+    switch ( this.save.dc.calculation ) {
+      case "custom":
+        this.save.dc.value = simplifyBonus(this.save.dc.formula, rollData);
+        break;
+      case "spellcasting":
+        ability = this.isSpell ? this.item.system.availableAbilities?.first() : null;
+        ability ??= this.actor?.system.attributes?.spellcasting;
+        break;
+      default:
+        ability = this.save.dc.calculation;
+        break;
+    }
```

Comment:
Is there some issue with calling into `this.ability` in `prepareFinalData`? Maybe some values might still change? If not, we should be able to remove some duplication here:

```suggestion
    let ability;
    if ( this.save.dc.calculation === "custom" ) this.save.dc.value = simplifyBonus(this.save.dc.formula, rollData);
    else ability = this.ability;
```

Reply:
No problem, just wrote `get ability` after this and never went back to simplify it :)